opm/simulators/aquifers/AquiferCarterTracy.hpp

```diff
@@ -223,8 +223,17 @@ class AquiferCarterTracy : public AquiferAnalytical<TypeTag>
              if (this->aquct_data_.initial_temperature.has_value())
                  temp = this->aquct_data_.initial_temperature.value();
 
-             Scalar rs = 0.0; // no dissolved CO2
-             Scalar waterViscosity = FluidSystem::oilPvt().viscosity(pvtRegionIdx(), temp, press, rs);
+             Scalar waterViscosity = 0.;
+             if (FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx)) {
+                 Scalar rs = 0.0; // no dissolved CO2
+                 waterViscosity = FluidSystem::oilPvt().viscosity(pvtRegionIdx(), temp, press, rs);
+             } else if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
+                 Scalar salt = 0.;
+                 Scalar rsw = 0.;
+                 waterViscosity = FluidSystem::waterPvt().viscosity(pvtRegionIdx(), temp, press, rsw, salt);
+             } else {
+                 OPM_THROW(std::runtime_error, "water or oil phase is needed to run CO2Store.");
```

Comment:
I tried to do some sanity checking in the constructor functions for AquiferInterface, but somehow `FluidSystem::phaseIsActive()` does not work there. Not sure why.